kernel/celltypes.h

```diff
@@ -56,7 +56,7 @@ struct CellTypes
 		setup_stdcells_mem();
 	}
 
-	void setup_type(RTLIL::IdString type, const pool<RTLIL::IdString> &inputs, const pool<RTLIL::IdString> &outputs, bool is_evaluable = false)
+	void setup_type(const RTLIL::IdString& type, const pool<RTLIL::IdString> &inputs, const pool<RTLIL::IdString> &outputs, bool is_evaluable = false)
```

Comment:
Are you sure those const references are worth it? AFAIK `IdString` is just an int.

Reply:
You are right. Something I did here got rid of a lot of calls to the (very expensive) `IdString` constructor that takes a string. Let me go back and figure out what. Maybe some implicit conversion somewhere. 

I'm thinking that it might be a good idea to mark that constructor as `explicit` (which is part of the Google style guide, BTW), since that is a performance pitfall in fairly innocent looking code.

Reply:
That sounds like something definitely worth investigating.

Reply:
Reverted. I'll investigate this separately.